geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/ClientSNIDropProxyAcceptanceTest.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.cache.client.NoAvailableLocatorsException;
+import org.apache.geode.cache.client.proxy.ProxySocketFactories;
+import org.apache.geode.test.junit.rules.IgnoreOnWindowsRule;
+
+public class ClientSNIDropProxyAcceptanceTest {
+
+  private static final URL DOCKER_COMPOSE_PATH =
+      ClientSNIDropProxyAcceptanceTest.class.getResource("docker-compose.yml");
+
+  // Docker compose does not work on windows in CI. Ignore this test on windows
+  // Using a RuleChain to make sure we ignore the test before the rule comes into play
+  @ClassRule
+  public static TestRule ignoreOnWindowsRule = new IgnoreOnWindowsRule();
+
+  @Rule
+  public DockerComposeRule docker = DockerComposeRule.builder()
+      .file(DOCKER_COMPOSE_PATH.getPath())
+      .build();
+
+  private ClientCache cache;
+
+  private String trustStorePath;
+
+  @Before
+  public void before() throws IOException, InterruptedException {
+    trustStorePath =
+        createTempFileFromResource(ClientSNIDropProxyAcceptanceTest.class,
+            "geode-config/truststore.jks")
+                .getAbsolutePath();
+    docker.exec(options("-T"), "geode",
+        arguments("gfsh", "run", "--file=/geode/scripts/geode-starter.gfsh"));
+
+  }
+
+  @Test
+  public void performSimpleOperationsDropSNIProxy()
+      throws IOException,
+      InterruptedException {
+    final Region<String, Integer> region = getRegion();
+
+    region.put("Roy Hobbs", 9);
+    assertThat(region.get("Roy Hobbs")).isEqualTo(9);
+
+    docker.containers()
+        .container("haproxy")
+        .stop();
+
+    assertThatThrownBy(() -> region.get("Roy Hobbs"))
+        .isInstanceOf(NoAvailableLocatorsException.class)
+        .hasMessageContaining("Unable to connect to any locators in the list");
+
+    docker.containers()
+        .container("haproxy")
+        .start();
+
+    assertThat(region.get("Roy Hobbs")).isEqualTo(9);
+
+    region.put("Bennie Rodriquez", 30);
+    assertThat(region.get("Bennie Rodriquez")).isEqualTo(30);
+
+    region.put("Jake Taylor", 7);
+    region.put("Crash Davis", 8);
+
+    region.put("Ricky Vaughn", 99);
+    region.put("Ebbie Calvin LaLoosh", 37);
+
+  }
+
+  public Region<String, Integer> getRegion() {
+    Properties gemFireProps = new Properties();
+    gemFireProps.setProperty(SSL_ENABLED_COMPONENTS, "all");
+    gemFireProps.setProperty(SSL_KEYSTORE_TYPE, "jks");
+    gemFireProps.setProperty(SSL_REQUIRE_AUTHENTICATION, "false");
+
+    gemFireProps.setProperty(SSL_TRUSTSTORE, trustStorePath);
+    gemFireProps.setProperty(SSL_TRUSTSTORE_PASSWORD, "geode");
+    gemFireProps.setProperty(SSL_ENDPOINT_IDENTIFICATION_ENABLED, "true");
+
+    int proxyPort = docker.containers()
+        .container("haproxy")
+        .port(15443)
+        .getExternalPort();
```

Comment:
without the bind of the port this will not match when the proxy is restarted

Reply:
outdated comment